dev/com.ibm.ws.dynacache/src/com/ibm/ws/cache/CacheConfig.java

```diff
@@ -377,6 +378,9 @@ public CacheConfig(Map<String, Object> map) {
         FieldInitializer.initFromSystemProperties(this);
         overrideCacheConfig(convert(map));
         determineCacheProvider();
+        _passedInProperties.putAll(map);
+        _passedInProperties.putAll(System.getProperties());
```

Comment:
When doing putAll across system properties you will need to synchronize on the properties object returned by System.getProperties while calling putAll here.